src/main/io/osd.c

```diff
@@ -1172,6 +1172,53 @@ static bool osdDrawSingleElement(uint8_t item)
             buff[5] = '\0';
             break;
         }
+
+    case OSD_CRUISE_HEADING_ERROR:
+        {
+            if (ARMING_FLAG(ARMED) && !FLIGHT_MODE(NAV_CRUISE_MODE)) {
+                displayWrite(osdDisplayPort, elemPosX, elemPosY, "     ");
+                return true;
+            }
+
+            buff[0] = SYM_HEADING;
+
+            if ((!ARMING_FLAG(ARMED)) || (FLIGHT_MODE(NAV_CRUISE_MODE) && isAdjustingPosition())) {
+                buff[1] = buff[2] = buff[3] = '-';
+            } else if (FLIGHT_MODE(NAV_CRUISE_MODE)) {
+                int16_t herr = lrintf(CENTIDEGREES_TO_DEGREES((float)navigationGetHeadingError()));
+                if (ABS(herr) > 99)
+                    strcpy(buff + 1, ">99");
+                else
+                    tfp_sprintf(buff + 1, "%3d", herr);
+            }
+
+            buff[4] = SYM_DEGREES;
+            buff[5] = '\0';
+            break;
+        }
+
+    case OSD_CRUISE_HEADING_ADJUSTMENT:
+        {
+            int16_t heading_adjust = lrintf(CENTIDEGREES_TO_DEGREES((float)getCruiseHeadingAdjustment()));
+
+            if (ARMING_FLAG(ARMED) && ((!FLIGHT_MODE(NAV_CRUISE_MODE)) || !(isAdjustingPosition() || isAdjustingHeading() || (heading_adjust != 0)))) {
+                displayWrite(osdDisplayPort, elemPosX, elemPosY, "      ");
```

Comment:
I don't like this extra call to displayWrite(). In functions as big as this one, I prefer to avoid early returns when possible. Can you make it write to buff the 5-space string? e.g.

```
            if (ARMING_FLAG(ARMED) && ((!FLIGHT_MODE(NAV_CRUISE_MODE)) || !(isAdjustingPosition() || isAdjustingHeading() || (heading_adjust != 0)))) {
                strcpy(buff, "     ");
            } else {
                buff[0] = SYM_HEADING;
                if (!ARMING_FLAG(ARMED))
                    buff[1] = buff[2] = buff[3] = buff[4] = '-';
                else if (FLIGHT_MODE(NAV_CRUISE_MODE)) {
                    tfp_sprintf(buff + 1, "%4d", heading_adjust);
                }
                buff[5] = SYM_DEGREES;
            }
            buff[6] = '\0';
```

Reply:
Just did a quick test and the early returns save 40 bytes of flash. Given the current state of flash in F3, I guess we're better off taking the flash savings.